libsolidity/analysis/PostTypeChecker.cpp

```diff
@@ -58,6 +58,9 @@ void PostTypeChecker::endVisit(ContractDefinition const&)
 	for (auto declaration: m_constVariables)
 		if (auto identifier = findCycle(declaration))
 			typeError(declaration->location(), "The value of the constant " + declaration->name() + " has a cyclic dependency via " + identifier->name() + ".");
+
+	m_constVariables.clear();
```

Comment:
My own thinking was that you cannot have cross contract cycles, because you cannot reference down the inheritance chain and libraries cannot reference back contracts.

I remember removing the check without clearing the state and getting some unintended results, like duplicated errors because IIRC the cycle would be found in both the child and the parent contract.